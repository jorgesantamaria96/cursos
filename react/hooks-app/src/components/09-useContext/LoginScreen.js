import React, { useContext } from "react";
import { UserContext } from "./UserContex";

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() =>
          setUser({
            id: 123,
            name: "Jorge",
          })
        }
      >
        Login
      </button>
    </div>
  );
};
