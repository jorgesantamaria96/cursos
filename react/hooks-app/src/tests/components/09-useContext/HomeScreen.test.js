import React from "react";
import { mount } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContex";

describe("Pruebas en el componente <HomeScreen />", () => {
  const user = {
    name: "Jorge",
    email: "jorge@jorge.com",
  };

  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomeScreen />
    </UserContext.Provider>
  );

  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
