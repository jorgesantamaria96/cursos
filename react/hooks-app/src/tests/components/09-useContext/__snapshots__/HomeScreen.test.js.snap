// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HomeScreen /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Jorge",
   "email": "jorge@jorge.com"
}
  </pre>
</div>
`;
